src/librustdoc/html/render.rs

```diff
@@ -877,12 +925,23 @@ impl DocFolder for Cache {
 
             match parent {
                 (parent, Some(path)) if is_method || (!self.privmod && !hidden_field) => {
+                    // Needed to determine `self` type.
+                    let parent_basename = if self.parent_stack.len() > 0 {
+                            match self.paths.get(&self.parent_stack[0]) {
+                                Some(&(ref fqp, _)) => Some(fqp[fqp.len() - 1].clone()),
+                                None => None
+                            }
+                        } else {
+                            None
+                        };
```

Comment:
I believe our typical form of indentation doesn't indent the body of the `if` over one extra (e.g. this should be tabbed left one).

I also think that this may be better expressed as `self.parent_stack.first().and_then(|..| { ... })` as it's already got the `Option` there.